src/main/java/com/aws/greengrass/deployment/DeploymentConfigMerger.java

```diff
@@ -147,6 +158,24 @@ private void updateActionForDeployment(Map<String, Object> newConfig, Deployment
             return;
         }
 
+        // As long as the timeout is not 0, we will try to run the connectivity validation
+        DeploymentConfigurationValidationPolicy policy = deployment.getDeploymentDocumentObj()
+                .getConfigurationValidationPolicy();
+        if (policy != null && Coerce.toInt(policy.timeoutInSeconds()) != 0) {
+            try (Context context = new Context()) {
+                ConnectivityValidator connectivityValidator = new ConnectivityValidator(kernel, context, mqttClient,
+                        factory, securityService, newConfig, deployment.getDeploymentDocumentObj().getTimestamp());
+                if (!connectivityValidator.validate(totallyCompleteFuture, deviceConfiguration)) {
+                    return;
+                }
+            } catch (Throwable e) {
+                logger.atInfo().cause(e).log("Unexpected exception during connectivity validation");
+                if (Thread.currentThread().isInterrupted()) {
+                    throw new InterruptedException("Connectivity validation task was interrupted");
+                }
+            }
```

Comment:
This is confusing - I understand the intention of catching any unexpected exception, but the handling is incorrect.

If any exception is thrown, you need to just fail the deployment.

Do not throw the InterruptedException out (which will never happen because you already handled it inside valiadate)

Do not swallow other exceptions - make sure you fail the deployment